src/index.js

```diff
@@ -13,10 +13,16 @@ export default (Base = HTMLElement) =>
       };
     }
     renderer(renderRoot, renderCallback) {
+      this._renderRoot = renderRoot
       this._preactDom = render(
         renderCallback(),
         renderRoot,
         this._preactDom || renderRoot.children[0]
       );
     }
+    disconnectedCallback() {
+      super.disconnectedCallback && super.disconnectedCallback();
+      this._preactDom = render(<div/>,this._renderRoot,this._preactDom);
```

Comment:
original is correctly replaced with empty div, although Preact won't call `componentWillUnmount` on mounted wrapped component ...

Reply:
Also it's not a shadowDOM thing. Following wont trigger willUnmount as well ( pure Preact )

```js
import { h, Component, render } from 'preact'
class App extends Component {
  render(){
    return <div>asd</div>
  }
  componentDidMount(){
    console.log('hi')
  }
  componentWillUnmount(){
    console.log('bye!')
  }
}
const mountTo = document.getElementById("root")
let root;

  root = render(<App />, mountTo);
  console.log(root)
  setTimeout(()=>{
   // element is replaced bud willUnmount won't be called ... Preact BUG ?
    render(<div>div</div>, mountTo, root)
  },1400)
```

Reply:
Yeah, I don't have any experience here. Might be worth pinging @developit about this? It looks like unmounting at a node is in compat, but I don't want to place a dependency on that https://github.com/developit/preact/issues/53.